build.rs

```diff
@@ -266,3 +269,61 @@ fn process_command_output(output: &Output) -> Result<()> {
         Ok(())
     }
 }
+
+fn build_error_code_reference() -> Result<()> {
+    let docs_path = Utf8PathBuf::from("./docs/source/errors.md");
+    let codes_dir = Utf8PathBuf::from("./src/error/metadata/codes");
+    let codes = fs::read_dir(codes_dir)?;
+
+    let mut all_descriptions = String::new();
+
+    // filter out Errs and non-file entries in the `/codes` dir
+    let code_files = codes
+        .into_iter()
+        .filter_map(Result::ok)
+        .filter(|e| !e.file_type().unwrap().is_dir());
```

Comment:
should we maybe error here instead?

Reply:
We definitely could 🤷‍♂️ Especially since this is at buildtime. I just left it as the simpler option in the meantime. Do you think I should change that?

Reply:
yeah since it's at build time i think maybe we should error here so we don't commit something that doesn't work, i imagine filtering out the errors could end up with some head scratches if something goes wrong

Reply:
I am going to keep this as is for now and we can get it to error in an xtask refactor.